server/builtin/app.go

```diff
@@ -242,16 +249,20 @@ func (a *builtinApp) Roundtrip(_ apps.App, creq apps.CallRequest, async bool) (o
 		if h.submitf == nil {
 			return nil, utils.ErrNotFound
 		}
-		return readcloser(h.submitf(creq))
+		return readcloser(h.submitf(ctx, creq))
 	}
 
 	return nil, utils.NewNotFoundError("%s does not handle %s", callPath, callType)
 }
 
-func (a *builtinApp) GetStatic(_ apps.App, path string) (io.ReadCloser, int, error) {
+func (a *builtinApp) GetStatic(_ context.Context, _ apps.App, path string) (io.ReadCloser, int, error) {
 	return nil, http.StatusNotFound, utils.NewNotFoundError("static support is not implemented")
 }
 
 func (a *builtinApp) newLocalizer(creq apps.CallRequest) *i18n.Localizer {
 	return a.conf.I18N().GetUserLocalizer(creq.Context.ActingUserID)
 }
+
+func (a *builtinApp) newContext(ctx context.Context, opts ...request.ContextOption) *request.Context {
```

Comment:
do we need to pass in `ctx`?

Reply:
Good catch. Fixed in 5cb282e09bee1d6ed7179831325e74b1b051d897.